template/template.go

```diff
@@ -76,3 +84,137 @@ func (t *Template) ExecuteHTMLString(html string, data interface{}) (string, err
 	err = tmpl.Execute(&buf, data)
 	return buf.String(), err
 }
+
+type FuncMap map[string]interface{}
+
+var DefaultFuncs = FuncMap{
+	"upper": strings.ToUpper,
+	"lower": strings.ToLower,
+	"title": strings.Title,
+	// pairs allows for in-order iteration of key/value pairs.
+	"pairs": func(m map[string]string) []Pair {
+		var (
+			pairs     = make([]Pair, 0, len(m))
+			keys      = make([]string, 0, len(m))
+			sortStart = 0
+		)
+		for k := range m {
+			if k == string(model.AlertNameLabel) {
+				keys = append([]string{k}, keys...)
+				sortStart = 1
+			} else {
+				keys = append(keys, k)
+			}
+		}
+		sort.Strings(keys[sortStart:])
+
+		for _, k := range keys {
+			pairs = append(pairs, Pair{k, m[k]})
+		}
+		return pairs
+	},
+	"firing": func(alerts []Alert) []Alert {
```

Comment:
We have that in one way: https://github.com/prometheus/alertmanager/blob/dev/notify/notify.go#L255-L257
Seems like this would make more sense to be configured per receiver configuration then?